cma/optimization_tools.py

```diff
@@ -197,26 +197,31 @@ def update(self, arx, xarchive=None, arf=None, evals=None):
         assert arf is not None
         # find failsave minimum
         try:
+            arf = np.squeeze(arf)
             minidx = np.nanargmin(arf)
         except ValueError:
             return
         if minidx is np.nan:
             return
-        minarf = arf[minidx]
+
+        minarf = arf[minidx] if arf.ndim else np.ascontiguousarray(arf)
```

Comment:
I don't quite understand this line. It seems that `minarf` is assigned to a scalar `if arf.ndim == 1` and to an array otherwise.

Reply:
Yes, the last call of the procedure returnes a one dimensional `arf` if the `tolfun` option is used.

But I just pushed a somewhat smoother version using `ascontinguousarray` earlier in the script, which makes both the `if...else` statements unnecessary.